server/sql.go

```diff
@@ -141,59 +141,52 @@ func (s *Server) registerSQLRoutes(g *echo.Group) {
 			return json.Marshal(rowSet)
 		}()
 
-		{
-			errMessage := ""
-			activityLevel := api.ActivityInfo
-			if err != nil {
-				errMessage = err.Error()
-				activityLevel = api.ActivityError
-			}
-
-			bytes, err := json.Marshal(api.ActivitySQLEditorQueryPayload{
-				Statement:    exec.Statement,
-				DurationNs:   time.Now().UnixNano() - start,
-				InstanceName: instance.Name,
-				DatabaseName: exec.DatabaseName,
-				Error:        errMessage,
-			})
-
-			if err != nil {
-				s.l.Warn("Failed to marshal activity after executing sql statement",
-					zap.String("database_name", exec.DatabaseName),
-					zap.String("instance_name", instance.Name),
-					zap.String("statement", exec.Statement),
-					zap.Error(err))
-				return echo.NewHTTPError(http.StatusInternalServerError, "Failed to construct activity payload").SetInternal(err)
-			}
-
-			activityCreate := &api.ActivityCreate{
-				CreatorID:   c.Get(getPrincipalIDContextKey()).(int),
-				Type:        api.ActivitySQLEditorQuery,
-				ContainerID: exec.InstanceID,
-				Level:       activityLevel,
-				Comment: fmt.Sprintf("Executed `%q` in database %q of instance %q.",
-					exec.Statement, exec.DatabaseName, instance.Name),
-				Payload: string(bytes),
-			}
-
-			_, err = s.ActivityManager.CreateActivity(ctx, activityCreate, &ActivityMeta{})
-
-			if err != nil {
-				s.l.Warn("Failed to create activity after executing sql statement",
-					zap.String("database_name", exec.DatabaseName),
-					zap.String("instance_name", instance.Name),
-					zap.String("statement", exec.Statement),
-					zap.Error(err))
-			}
-		}
-
 		resultSet := &api.SQLResultSet{}
 		if err == nil {
 			resultSet.Data = string(bytes)
 			s.l.Debug("Query result",
 				zap.String("statement", exec.Statement),
 				zap.String("data", resultSet.Data),
 			)
+
+			// Only record the history of successfully executed queries.
```

Comment:
If the query execution activities are for auditing purpose, we might want to record all executions but including its executions status.

Reply:
That's a good idea. If we have auditing center one day, it can show all the records about who did what.

Reply:
@tianzhou wdyt

Reply:
SGTM